core/src/main/java/tc/oc/pgm/api/filter/ReactorBased.java

```diff
@@ -0,0 +1,49 @@
+package tc.oc.pgm.api.filter;
+
+import org.bukkit.event.Event;
+import tc.oc.pgm.api.match.Match;
+import tc.oc.pgm.filters.dynamic.FilterMatchModule;
+import tc.oc.pgm.filters.dynamic.Filterable;
+
+/**
+ * Implemented by filters which can act dynamically but needs more than Bukkit {@link Event}s to
+ * properly invalidate {@link Filterable}s. A filter can theoretically both depend on listening to
+ * some events AND have a reactor, but this is rarely the case.
+ *
+ * <p>Often(But not necessarily always), in addition to using non-event factors when taking into
+ * account when to invalidate something, a filter implementing this will in wrap another dynamic
+ * filter and react based on the changed state of that filter.
+ */
+public interface ReactorBased extends FilterDefinition {
```

Comment:
based, based on what?

Call this ReactorFactory or something else, "Based" is a bit of a weird term to use here

Reply:
I guess it makes more sense in Norwegian. Based as in "basert (på) reaktor".